### PR TITLE
Feature/quiz

### PR DESCRIPTION
**Pull request by Bluesmile82**
Quizzes on the Key concepts page

https://vizzuality.atlassian.net/browse/NCP-125?focusedCommentId=31663&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-31663

https://www.figma.com/design/R3bsPj03L8Rx3D36hHIBTa/NCP-Designs-%5BInternal%5D?node-id=3612-4809&t=wQYVLsMcR7T2vTtS-1

### CODE REVIEW

File: src/app/key-concepts/(index)/page.tsx [hunk 1]
```diff
@@ -39,6 +41,140 @@ const LightBulbMessage = ({ children }: { children: React.ReactNode }) => (
...
+          </>
+        ),
+        options: ["Renewable Resources", "Not-Renewable Resources", "Services"],
+        answer: 1,
+      },
+      {
+        title: (
+          <>
+            How can <span className="text-orange-500">business&apos;s actions</span> affect the
+            environment?
```
Comment (clementprdhomme) [inline src/app/key-concepts/(index)/page.tsx]: I think the highlighting for this question has changed.

File: src/app/key-concepts/(index)/page.tsx [hunk 2]
```diff
@@ -39,6 +41,140 @@ const LightBulbMessage = ({ children }: { children: React.ReactNode }) => (
...
+          "Indirectly through supply chains",
+          "Both directly and indirectly",
+        ],
+        answer: 2,
+      },
+      {
+        title: (
+          <>
+            <span className="text-orange-500">Climate change </span>
+            is an example of which type of impact?
```
Comment (clementprdhomme) [inline src/app/key-concepts/(index)/page.tsx]: Same here. The highlighting has changed.

File: src/components/quiz/index.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                height="109"
+              />
+            </div>
+          )
+        )}
+      </div>
+      <div className="text-sm mt-6 flex w-full flex-col-reverse items-center justify-between gap-4 border-t border-dashed border-t-grey-400 px-6 pb-10 pt-6 text-white lg:mt-0 lg:h-[84px] lg:flex-row lg:px-[53px] lg:py-[30px]">
+        We won’t use this collected data in any way, except having an overview of the content
+        experience.
+        {waitForNext && quizQuestion < 2 && (
```
Comment (clementprdhomme) [inline src/components/quiz/index.tsx]: There should be a previous button as well: https://www.figma.com/design/R3bsPj03L8Rx3D36hHIBTa/NCP-Designs-%5BInternal%5D?node-id=3373-7850&t=AXgvlM4MyxpDd9e8-4

File: src/components/quiz/index.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+          <h3 className="text-2xl tracking-tight lg:text-4.5xl">{title}</h3>
+          <div className="flex flex-wrap gap-5">
+            {options.map((option, index) => (
+              <QuizButton
+                key={`${key}-${index}`}
+                onMouseEnter={() => setHoveredOption(index)}
+                onMouseLeave={() => setHoveredOption(null)}
+                isRight={answer === index}
+                isActive={selectedOption === index}
+                onClick={() => {
```
Comment (clementprdhomme) [inline src/components/quiz/index.tsx:78]: If the user clicks on a wrong answer, the correct answer should also be highlighted: https://www.figma.com/design/R3bsPj03L8Rx3D36hHIBTa/NCP-Designs-%5BInternal%5D?node-id=3612-4869&t=AXgvlM4MyxpDd9e8-4

File: src/components/quiz/index.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        <div className="z-10 mx-6 flex flex-col gap-6 text-white lg:ml-[53px] lg:mr-0 lg:max-w-[835px] lg:gap-8 lg:pb-[73px]">
+          <div className="flex flex-col gap-2 lg:flex-row lg:gap-5">
+            <span className="flex items-center gap-3">
+              Check your Understanding
+              <span className="h-px flex-1 bg-white lg:hidden"></span>
+            </span>
+            <div className="hidden text-grey-400 lg:block">|</div>
+            <span className={cn("flex gap-1 text-white", colorClass)}>
+              <span className="hidden lg:block">
+                {chapterNumber}-0{quizQuestion + 1}
```
Comment (clementprdhomme) [inline src/components/quiz/index.tsx]: I don't think the first number is the chapter number. If you look at the design, it seems like it's the number of question. See: https://www.figma.com/design/R3bsPj03L8Rx3D36hHIBTa/NCP-Designs-%5BInternal%5D?node-id=3392-9042&t=AXgvlM4MyxpDd9e8-4.

File: src/components/quiz/answered-animation-mobile.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  visible,
+  isRight = false,
+}: {
+  visible: boolean;
+  isRight: boolean;
+}) => {
+  return createPortal(
+    <AnimatePresence>
+      {visible && (
+        <motion.div
```
Comment (clementprdhomme) [inline src/components/quiz/answered-animation-mobile.tsx:15]: I don't think this overlay should be full screen. Shouldn't it just cover the quiz container?

File: src/components/quiz/index.tsx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    setWaitForNext(false);
+    setQuizQuestion(quizQuestion + 1);
+  };
+
+  const isRight = selectedOption === answer;
+  return (
+    <div className="flex w-full max-w-7xl flex-col overflow-hidden bg-black pt-10 lg:mb-20 lg:pt-[77px]">
+      <AnsweredAnimationMobile isRight={isRight} visible={playMobileAnimation} />
+      <div className="relative mb-[15px] flex w-full justify-between">
+        <div className="z-10 mx-6 flex flex-col gap-6 text-white lg:ml-[53px] lg:mr-0 lg:max-w-[835px] lg:gap-8 lg:pb-[73px]">
```
Comment (clementprdhomme) [inline src/components/quiz/index.tsx]: I think that when the mobile layout is used, the white line should extend until the end of the container. Don't you think?
The white line after “Check your Understanding” doesn't reach the end of the container

File: src/components/quiz/index.tsx (hunk 3 above)
Comment (Bluesmile82) [reply src/components/quiz/index.tsx]: Teona decided not to include it

**Comment by Bluesmile82** [comment]
@clementprdhomme, Teona decided to remove the images as we didn't have them for all the quizzes and were causing some positioning problems with the text. Also the answer validation message now appears and fades out always. And the lower text was changed

**Review by clementprdhomme — APPROVED** [review]
I think the animation is too short to let the user read the text but apart from that, everything's ok.

**Pull request merged**